Yarn upgrade of @storybook/react 6.4.0 to version 6.5.16

jentfoo opened the pull request:
Storybook has a large number of transitive dependencies (as can be seen in the change of this commit). This update will help address some remaining Dependabot alerts that were not fully satisfied earlier:
* https://github.com/gravitational/teleport/security/dependabot/45 (dependency removed)
* https://github.com/gravitational/teleport/security/dependabot/52 (dependency removed)
* https://github.com/gravitational/teleport/security/dependabot/57 (vulnerable version removed)

I worry this may have more risk than previous upgrades because of the dependencies removed, but I am not seeing any errors in a build or a manual smoke test of the UI.

jentfoo committed:
Yarn upgrade of @storybook/react to 6.5.16

This dependency has a large number of transitive dependencies (as can be seen in the change of this commit).
This will help address some remaining Dependabot alerts that were not fully satisfied earlier:
* https://github.com/gravitational/teleport/security/dependabot/45 (dependency removed)
* https://github.com/gravitational/teleport/security/dependabot/52 (dependency removed)
* https://github.com/gravitational/teleport/security/dependabot/57 (vulnerable version removed)

zmb3 commented:
If we want to be safe, we can hold this in master so that it's part of the v13 test plan in a few months.

jentfoo committed:
Pin x-default-browser version to 0.5.2

This dependency was introduced from `@storybook/core-server@6.5.16`, but unfortunately depends on a vulnerable version of `trim-newlines`.

Updating x-default-browser results in an upgrade of `default-browser-id`.  The new version of `default-browser-id` no longer pulls in `meow`.  This avoids the vulnerable `trim-newlines` but also removes a good number of additional dependencies.

jentfoo commented:
@zmb3 I think that is a good plan due to the amount of change I have been introducing

jentfoo committed:
Update storybook/react defined version in web/packages/build/package.json
